### ISSUE TITLE
[Win] CAN Wrapper

### ISSUE THREAD

**Issue by AndreasFreund** [amosproj/amos2024ss07-updating-flash-boot-loader#12]
## User story
1. As a developer
2. I want to send CAN messages using a Vector Adapter without dealing with hardware directly
3. So that my code stays portable / testable

## Acceptance criteria
* .hpp file defining platform agnostic methods / interface / class for sending CAN messages
* .hpp file defining specific implementation for Vector Adapter
* .cpp file implementing vector adapter driver
* simple demo app (CLI) sending a message and receiving a message (e.g. from devkit, some kind of loopback mode, or vector virtual can bus)

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release
